segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataCleaner.java

```diff
@@ -179,6 +180,26 @@ protected void doStop() {
                 }, this.executor);
     }
 
+    private CompletableFuture<Void> deleteUnusedTransientSegments() {
+        long lastSeqNo = this.lastIterationSequenceNumber.getAndSet(this.metadata.getOperationSequenceNumber());
+        long traceId = LoggerHelpers.traceEnterWithContext(log, this.traceObjectId, "deleteUnusedTransientSegments", lastSeqNo);
+
+        // Serialize only those segments that are still alive (not deleted or merged - those will get removed anyway).
+        this.metadata.getEvictionCandidates(lastSeqNo, this.config.getMaxConcurrentSegmentEvictionCount())
+                .stream()
+                .filter(sm -> sm.getType() == SegmentType.TRANSIENT_SEGMENT && sm.isDeleted())
+                .forEach(sm -> {
+                    // Todo: AppendProcessor - isTransientSegmentActive()?
+                });
+
+        LoggerHelpers.traceLeave(log, this.traceObjectId, "deleteUnusedTransientSegments", traceId);
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private CompletableFuture<Void> runOnceInternal() {
+        return evictUnusedSegments().thenAccept(unused -> deleteUnusedTransientSegments());
```

Comment:
`thenCompose`

(`thenAccept` will just initiate, but not await `deleteUnusedTransientSemgents` to complete - please refer to the `CompletableFuture` JavaDoc for details).